### PR TITLE
Add initial Import step to New Experiment modal

### PR DESCRIPTION
**Pull request by Auz**

### CODE REVIEW

File: packages/front-end/components/Experiment/ImportExperimentList.tsx [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+  if (!data || !ready) {
+    return <LoadingOverlay />;
+  }
+
+  return (
+    <>
+      <h2>Import Experiments</h2>
+      <p>
+        From datasource:{" "}
+        <strong>{getDatasourceById(data.experiments.datasource).name}</strong>
```
Comment (jdorn) [inline packages/front-end/components/Experiment/ImportExperimentList.tsx:69]: `getDatasourceById` might return null

File: packages/front-end/components/Experiment/ImportExperimentList.tsx [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+      </p>
+      <div className="row mb-3">
+        <div className="col-auto">
+          <form
+            onSubmit={async (e) => {
+              console.log("import exp form submitted");
```
Comment (jdorn) [inline packages/front-end/components/Experiment/ImportExperimentList.tsx:75]: Remove console.log

File: packages/front-end/components/Experiment/ImportExperimentList.tsx [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+                                },
+                              ],
+                              // Default to stopped if the last data was more than 3 days ago
+                              status:
+                                getValidDate(e.endDate).getTime() <
+                                Date.now() - 72 * 60 * 60 * 1000
+                                  ? "stopped"
+                                  : "running",
+                            };
+                            console.log("firing onimport");
```
Comment (jdorn) [inline packages/front-end/components/Experiment/ImportExperimentList.tsx:214]: Remove console.log

File: packages/front-end/pages/datasources/[did].tsx [hunk 4]
```diff
@@ -32,7 +32,7 @@ const DataSourcePage: FC = () => {
     error,
   } = useDefinitions();
   const { did } = router.query as { did: string };
-
+  console.log(did);
```
Comment (jdorn) [inline packages/front-end/pages/datasources/[did].tsx:35]: Revert this

File: packages/front-end/components/Experiment/NewExperimentForm.tsx [hunk 5]
```diff
@@ -24,6 +24,8 @@ import Field from "../Forms/Field";
 import { getValidDate } from "../../services/dates";
 import { GBAddCircle } from "../Icons";
 import SelectField from "../Forms/SelectField";
+import ImportExperimentList from "./ImportExperimentList";
```
Comment (jdorn) [inline packages/front-end/components/Experiment/NewExperimentForm.tsx:27]: I think we should take a different approach.

1.  Revert all changes in this `NewExperimentForm` component (already really hard to maintain)
2.  Make a new `ImportExperimentModal` component:
     ```tsx
     const [selected, setSelected] = useState<null|Partial<ExperimentInterfaceStringDates>>(null);

     if(selected) {
       return <NewExperimentForm initialValue={selected} .../>
     }

     return (
       <Modal ...>
         <ImportExperimentList
            ...
            onImport={(exp) => setSelected(exp)}
         />
       </Modal>
     )
     ```

Comment (Auz) [reply packages/front-end/components/Experiment/NewExperimentForm.tsx:27]: I can do this, but there would be no way to quickly switch between a new exp and an import. Do you think this is okay?

File: packages/front-end/components/Experiment/ImportExperimentModal.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  const [datasourceId, setDatasourceId] = useState(datasources?.[0]?.id);
+  const [importId, setImportId] = useState(null);
+
+  const { apiCall } = useAuth();
+
+  const getImportId = async () => {
+    const res = await apiCall<{ id: string }>("/experiments/import", {
+      method: "POST",
+      body: JSON.stringify({
+        datasource: datasourceId,
```
Comment (jdorn) [inline packages/front-end/components/Experiment/ImportExperimentModal.tsx:31]: datasourceId might be undefined.  We shouldn't do an API call in that case.

File: packages/front-end/components/Experiment/ImportExperimentModal.tsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    });
+    if (res?.id) {
+      setImportId(res.id);
+    }
+  };
+  useEffect(() => {
+    getImportId();
+  }, [datasourceId]);
+
+  if (selected || !importModal) {
```
Comment (jdorn) [inline packages/front-end/components/Experiment/ImportExperimentModal.tsx:42]: We should also show NewExperimentForm if datasourceId is null

Comment (Auz) [reply packages/front-end/components/Experiment/ImportExperimentModal.tsx]: done

File: packages/front-end/components/Experiment/ImportExperimentModal.tsx (hunk 6 above)
Comment (Auz) [reply packages/front-end/components/Experiment/ImportExperimentModal.tsx]: done

**Pull request merged**